app/scripts/controllers/alert.js

```diff
@@ -14,38 +18,35 @@ import ObservableStore from 'obs-store'
  * @property {AlertControllerInitState} initState - The initial controller state
  */
 
-export const ALERT_TYPES = {
-  unconnectedAccount: 'unconnectedAccount',
-  // enumerated here but has no background state
-  invalidCustomNetwork: 'invalidCustomNetwork',
-}
-
 const defaultState = {
-  alertEnabledness: Object.keys(ALERT_TYPES).reduce(
+  alertEnabledness: TOGGLEABLE_ALERT_TYPES.reduce(
     (alertEnabledness, alertType) => {
       alertEnabledness[alertType] = true
       return alertEnabledness
     },
     {},
   ),
   unconnectedAccountAlertShownOrigins: {},
+  web3ShimUsageOrigins: {},
 }
 
 /**
- * Controller responsible for maintaining
- * alert related state
+ * Controller responsible for maintaining alert-related state.
  */
 export default class AlertController {
   /**
    * @constructor
    * @param {AlertControllerOptions} [opts] - Controller configuration parameters
    */
   constructor(opts = {}) {
-    const { initState, preferencesStore } = opts
+    const { initState = {}, preferencesStore } = opts
     const state = {
-      ...defaultState,
-      ...initState,
+      alertEnabledness: {
+        ...defaultState.alertEnabledness,
+        ...initState.alertEnabledness,
+      },
       unconnectedAccountAlertShownOrigins: {},
+      web3ShimUsageOrigins: {},
```

Comment:
Q: should you just spread `...defaultState` before `alertEnabledness` and remove the empty objects here? That way if default state changes in the future it is easily extensible here? No strong opinions on this, it is safe the way it is.

Reply:
actually, in retrospect -- do we want persistence for `web3ShimUsageOrigins`? Then we'd want to spread `initState.web3ShimUsageOrigins` into the state declaration

Reply:
We don't want to persist `web3ShimUsageOrigins`, because we want dismissed web3 shim usage notifications to reappear if the extension is restarted (assuming the notification is enabled), and storing the origins could contribute to state bloat over time.

Nevertheless, I followed your suggestion about spreading `defaultState` first, it's better!